configuration/postgres/postgres.go

```diff
@@ -0,0 +1,415 @@
+		m.connectionString = val
+	} else {
+		return m, fmt.Errorf(ErrorMissingConnectionString)
+	}
+
+	if tbl, ok := cmetadata.Properties[configtablekey]; ok && tbl != "" {
+		if !allowedChars.MatchString(tbl) {
+			return m, fmt.Errorf("invalid table name : '%v'. non-alphanumerics are not supported", tbl)
+		}
+		if len(tbl) > maxIdentifierLength {
+			return m, fmt.Errorf(ErrorTooLongFieldLength+" - tableName : '%v'. max allowed field length is %v ", tbl, maxIdentifierLength)
+		}
+		m.configTable = tbl
+	} else {
+		return m, fmt.Errorf(ErrorMissingTableName)
+	}
+
+	// configure maxTimeout if provided
+	if mxTimeout, ok := cmetadata.Properties[connMaxIdleTimeKey]; ok && mxTimeout != "" {
+		if t, err := time.ParseDuration(mxTimeout); err == nil {
+			m.maxIdleTime = t
+		} else {
+			return m, fmt.Errorf(ErrorMissinMaxTimeout)
+		}
+	}
+	return m, nil
+}
+
+func Connect(ctx context.Context, conn string, maxTimeout time.Duration) (*pgxpool.Pool, error) {
+	pool, err := pgxpool.Connect(ctx, conn)
+	if err != nil {
+		return nil, fmt.Errorf("postgres configuration store connection error : %s", err)
+	}
+	pingErr := pool.Ping(ctx)
+	if pingErr != nil {
+		return nil, fmt.Errorf("postgres configuration store ping error : %s", pingErr)
+	}
+	return pool, nil
+}
+
+func buildQuery(req *configuration.GetRequest, configTable string) (string, []interface{}, error) {
+	var query string
+	var params []interface{}
+	if len(req.Keys) == 0 {
+		query = "SELECT * FROM " + configTable
+	} else {
+		var queryBuilder strings.Builder
+		queryBuilder.WriteString("SELECT * FROM " + configTable + " WHERE KEY IN (")
+		var paramWildcard []string
+		paramPosition := 1
+		for _, v := range req.Keys {
+			paramWildcard = append(paramWildcard, "$"+strconv.Itoa(paramPosition))
+			params = append(params, v)
+			paramPosition++
+		}
+		queryBuilder.WriteString(strings.Join(paramWildcard, " , "))
+		queryBuilder.WriteString(")")
+
+		query = queryBuilder.String()
+
+		if len(req.Metadata) > 0 {
+			var s strings.Builder
+			i, j := len(req.Metadata), 0
+			s.WriteString(" AND ")
+			for k, v := range req.Metadata {
+				temp := "$" + strconv.Itoa(paramPosition) + " = " + "$" + strconv.Itoa(paramPosition+1)
+				s.WriteString(temp)
+				params = append(params, k, v)
+				paramPosition += 2
+				if j++; j < i {
+					s.WriteString(" AND ")
+				}
+			}
+			query += s.String()
+		}
+	}
+	return query, params, nil
+}
+
+func (p *ConfigurationStore) isSubscriptionActive(req *configuration.SubscribeRequest) (string, bool) {
+	for _, trigger := range req.Metadata {
+		for key2, sub := range p.ActiveSubscriptions {
+			if res := strings.EqualFold(trigger, key2); res {
+				return sub.uuid, true
+			}
+		}
+	}
+	return " ", false
+}
+
+func (p *ConfigurationStore) isSubscribed(trigger string, key string, subscription string) bool {
+	for t, s := range p.ActiveSubscriptions {
+		if t == trigger && s.uuid == subscription {
```

Comment:
Replace with slices.Contains? https://pkg.go.dev/golang.org/x/exp/slices#Contains

Reply:
@berndverst @ItalyPaleAle any issue with using this method?

Reply:
None from me, we have been using generics already in Dapr.

Reply:
done